### ISSUE TITLE
finetuning action recognition with my own data

### ISSUE THREAD

**Issue by skelder** [dmlc/gluon-cv#1258]
Hello, I apologize if the question is too trivial.

However, when I try to use the train_recognizer.py script on my own data-set I have weird behavior.

The accuracy starts with value around 12-14 and stays that way for the whole training sequence, with loss staying around 2.

flags i tried to modify
pretraining 
input-5d 
mode
none of them changed the behavior in meaningful way

`python train_recognizer.py --dataset custom --data-dir ~/datasetmaking/myWork/Crop/Recordings --val-data-dir ~/datasetmaking/myWork/Crop/Recordings --train-list ~/datasetmaking/myWork/train.txt --val-list ~/datasetmaking/myWork/val.txt --num-gpus 2 --num-epochs 100 --model i3d_resnet50_v1_custom --num-classes 8 --video-loader --use-decord --new-length 32 --lr 0.001 --batch-size 1 --mode hybrid --use-pretrained`

this is the last command i used.

i think my data is the main problem here, I have total 800 videos of 8 different actions with 640 used for training.

also I wanted to clarify about input_size. Is it height and width of the video?
default says 224 so my videos should be 224x224 pixels?

**Comment by bryanyzhu** [comment]
Hi, thank you for using GluonCV. In your command, I think you can remove `--use-pretrained` and try again. Because if you use `--use-pretrained`, you are using an ImageNet pretrained weights. If you don't use it, you are using a Kinetics400 pretrained weights, that may give you better performance. You can find more details [here](https://github.com/dmlc/gluon-cv/blob/master/gluoncv/model_zoo/action_recognition/i3d_resnet.py#L1246).

Another thing is your batch size is too small. Batch normalization only works when batch size is large (at least 4 or 8). 

The input size is the size going to the network, which is 224x224. But your raw video is not necessarily to be 224x224 pixels. In our data pre-processing pipeline, we will first resize your video to 256x340, and randomly crop a 224x224 patch to feed into the network. 

Hope this helps.

**Comment by skelder** [comment]
Hi again,

Thanks to your tips i managed to get the performance up to around 80% on the i3d network.

Now i am trying to do the same with the slowfast.

One thing i am noticing right now with slowfast_4x16_resnet50_custom is that while training the accuracy will go up for 4-5 epochs and then drop for about 15-20% and then start going up again. 
Thankfully it reaches higher accuracy in about 5 epochs before dropping again.

I am just curious whether it is an expected behavior or does this indicate something?

FYI: I took a look at the log file for slowfast_4x16_resnet50_kinetics400 in the model zoo and used the same settings for calling the train_recognizer.py

`Namespace(accumulate=1, batch_norm=False, batch_size=8, clip_grad=0, crop_ratio=0.875, data_dir='/home/kky2927/datasetmaking/myWork/Crop/Recordings/', dataset='custom', dtype='float32', eval=False, fast_temporal_stride=2, freeze_bn=False, hard_weight=0.5, hashtag='', input_5d=False, input_size=224, kvstore=None, label_smoothing=False, last_gamma=False, log_interval=50, logging_file='train.log', lr=0.11, lr_decay=0.1, lr_decay_epoch='40,60', lr_decay_period=0, lr_mode='cosine', mixup=False, mixup_alpha=0.2, mixup_off_epoch=0, mode='hybrid', model='slowfast_4x16_resnet50_custom', momentum=0.9, new_height=256, new_length=64, new_step=1, new_width=340, no_wd=False, num_classes=8, num_crop=1, num_epochs=200, num_gpus=2, num_segments=1, num_workers=32, partial_bn=False, prefetch_ratio=1.0, resume_epoch=0, resume_params='', resume_states='', save_dir='params', save_frequency=10, scale_ratios='1.0, 0.875, 0.75, 0.66', slow_temporal_stride=16, slowfast=True, teacher=None, temperature=20, train_list='/home/kky2927/datasetmaking/myWork/train.txt', use_amp=False, use_decord=True, use_gn=False, use_pretrained=False, use_se=False, use_tsn=False, val_data_dir='/home/kky2927/datasetmaking/myWork/Crop/Recordings/', val_list='/home/kky2927/datasetmaking/myWork/val.txt', video_loader=True, warmup_epochs=24, warmup_lr=0.01, wd=0.0001)`

**Comment by bryanyzhu** [comment]
Yes, this is normal behavior.

For example, when we train SlowFast on Kinetics400, the performance first goes up till around epoch 20, then it starts to drop (about 2~3 percent) till epoch 50, then it goes up again. 

So as long as you are satisfied with your final performance, you are good to go.

**Comment by skelder** [comment]
Thanks a lot!
